File: 1.swift
func findMaxSum(_ s: String) -> String {
    var maxSum = 0
    var result = ""
    var currentSum = 0
    var currentSub = ""
    var hasDigits = false
    
    for char in s {
        if char.isNumber {
            hasDigits = true
            currentSub.append(char)
            currentSum += Int(String(char)) ?? 0
        } else {
            if !currentSub.isEmpty {
                if currentSum > maxSum {
                    maxSum = currentSum
                    result = currentSub
                }
                currentSub = ""
                currentSum = 0
            }
        }
    }
    
    if !currentSub.isEmpty && currentSum > maxSum {
        result = currentSub
    }
    
    return hasDigits ? result : "В строке нет цифр"
}

// Тестовые примеры
print(findMaxSum("abc1234def56gh789")) // "789"
print(findMaxSum("a1b2"))              // "2"
print(findMaxSum("HelloWorld!"))       // "В строке нет цифр"

File: 1.js
function findMaxSum(s) {
    let maxSum = 0;
    let result = '';
    let currentSum = 0;
    let currentSub = '';
    let hasDigits = false;

    for (const char of s) {
        if (/\d/.test(char)) {
            hasDigits = true;
            currentSub += char;
            currentSum += Number(char);
        } else {
            if (currentSub) {
                if (currentSum > maxSum) {
                    maxSum = currentSum;
                    result = currentSub;
                }
                currentSub = '';
                currentSum = 0;
            }
        }
    }
// Проверяем последнюю подстроку
    if (currentSub) {
        if (currentSum > maxSum) {
            result = currentSub;
        }
    }

    return hasDigits ? result : 'В строке нет цифр';
}

// Примеры использования
console.log(findMaxSum("abc1234def56gh789")); // "789" (сумма 24)
console.log(findMaxSum("a1b22c333d4444"));    // "4444" (сумма 16)
console.log(findMaxSum("HelloWorld!"));       // "В строке нет цифр!"
